examples/infra/mod.rs

```diff
@@ -450,15 +444,19 @@ where
             NetworkNodeType::Regular
         };
         let node_index = config.node_index;
-        let bound_addr = format!(
+        let port_index = match libp2p_config.index_ports {
+            true => node_index,
+            false => 0,
+        };
+        let bound_addr: Multiaddr = format!(
             "/{}/{}/tcp/{}",
             if libp2p_config.public_ip.is_ipv4() {
                 "ip4"
             } else {
                 "ip6"
             },
             libp2p_config.public_ip,
-            libp2p_config.base_port + node_index as u16
+            libp2p_config.base_port as u64 + port_index
```

Comment:
Do we really want to allow an invalid port range like this? I'm assuming this is because of the lack of u16 support in the config specification?

Reply:
I'll update this in the PR for the sequencing examples